test/t/float.rb

```
@@ -158,6 +158,65 @@
+  assert_true(a.equal?(a))
+  assert_true(a.equal?(c))          # the same object, passed around
+  assert_false(a.equal?(b))         # two NaNs, made apart
+  assert_false(a.equal?(Float::NAN))
+
+  assert_equal(0, [a].index(a))
+  assert_nil([a].index(b))
+  assert_true([a].include?(a))
+  assert_false([a].include?(b))
+  assert_equal(1, [a].count(a))
+  assert_equal(0, [a].count(b))
+  assert_true([1.0, a] == [1.0, a])
```

RuboCop (1.89.0)

[warning] Binary operator `==` has identical operands.

(Lint/BinaryOperatorWithIdenticalOperands)